### ISSUE TITLE
httpProvider on StreamClient.builder() seems to do nothing

### ISSUE THREAD

**Issue by MikeyCarter** [oracle/oci-java-sdk#488]
I have to specify httpProvider in my setup or I get "No http provider available; add dependency on one of the oci-java-sdk-common-httpclient-* choices, e.g. oci-java-sdk-common-httpclient-jersey"

With the StreamClient it seams to work differently than other clients and it comes up with that error even with httpProvider defined.  Doing a trace it goes to the "defaultProvider" instead of using what I specified.

**Comment by KartikShrikantHegde** [comment]
Hi @MikeyCarter , thanks for notifying the issue. One of our engineers will take a look. Thanks!

**Comment by mricken** [comment]
Hi @MikeyCarter , I was able to successfully run this example: https://github.com/oracle/oci-java-sdk/blob/master/bmc-examples/src/main/java/StreamsExample.java

Can you show me how you are creating the client? You say "you have to specify httpProvider", which makes me think it's different than just

```
StreamClient streamClient = StreamClient.builder().stream(stream).build(provider);
```

Thanks!

**Comment by MikeyCarter** [comment]
What I'm assuming is going on is the default build() is calling one of these, which is dropping the builder you supply for a new one.

```
    @Deprecated
    public StreamClient(
            com.oracle.bmc.auth.AbstractAuthenticationDetailsProvider authenticationDetailsProvider,
            com.oracle.bmc.ClientConfiguration configuration,
            com.oracle.bmc.http.ClientConfigurator clientConfigurator,
            com.oracle.bmc.http.signing.RequestSignerFactory defaultRequestSignerFactory,
            java.util.List<com.oracle.bmc.http.ClientConfigurator> additionalClientConfigurators) {
        this(
                builder()
                        .configuration(configuration)
                        .clientConfigurator(clientConfigurator)
                        .requestSignerFactory(defaultRequestSignerFactory)
                        .additionalClientConfigurators(additionalClientConfigurators),
                authenticationDetailsProvider);
    }
```

**Comment by mricken** [comment]
Thank you for the additional information, @MikeyCarter . I'm now able to reproduce the problem and can work on a solution. Thanks for bringing this to our attention!

**Comment by mricken** [comment]
A fix for this problem should be released on April 11.

**Comment by mricken** [comment]
This was fixed in version 3.11.0.

@MikeyCarter , could you please give it a try, and if it doesn't work, re-open this issue? Thanks again for bringing this to our attention.
